rclcpp/src/rclcpp/executor.cpp

```diff
@@ -232,14 +247,20 @@ Executor::spin_some(std::chrono::nanoseconds max_duration)
     throw std::runtime_error("spin_some() called while already spinning");
   }
   RCLCPP_SCOPE_EXIT(this->spinning.store(false); );
-  // non-blocking call to pre-load all available work
-  wait_for_work(std::chrono::milliseconds::zero());
+  bool work_available = false;
```

Comment:
Looking through the logic of this, I think that this variable could be named better.  In particular, I think this variable really controls whether to "check if there is new work available".  Is that correct?  If so, I'd suggest `check_for_work_available`, or something like that, and invert the logic.

Reply:
mmm, `work_available` sounds like a better name to me.
`work_available` is `true` if the last `wait_for_work` resulted in at least one available entity.

We're actually checking for more work (i.e. calling `wait_for_work`) when `work_available` is `false`, so the new name sounds confusing (except if I invert the logic).

Reply:
I have no strong preference here.

Reply:
Yeah, you'd have to to invert the logic for the new name to make sense.

The main reason I was pushing for a change is that it is somewhat subtle how `spin_some` differs from `spin_all`, and it took a good 10 minutes of staring at the code for me to understand the difference.  But I don't feel very strongly about this one, so if you think it is clearer as-is, that's fine with me.

Reply:
Yes, I think that the difference between both methods is subtle and might be a bit confusing.
With the last doc block update, I hope it will be clear enough for users.

About naming of that variable, I think that depending on the person one name or the other may be clearer. As I don't see a clear benefit, I will keep the original name.